src/scale.c

```diff
@@ -135,15 +121,13 @@ avifBool avifImageScale(avifImage * image, uint32_t dstWidth, uint32_t dstHeight
     }
 
     if (srcAlphaPlane) {
-        const int fixedPointDivWidth = (int)(((int64_t)(srcWidth) << 16) / dstWidth);
-        const int64_t maxFixedValueWidth = (int64_t)fixedPointDivWidth * dstWidth;
-        if (maxFixedValueWidth > INT_MAX) {
+        // A simple conservative check to avoid integer overflows in libyuv's ScalePlane() and
+        // ScalePlane_12() functions.
+        if (srcWidth > 16384) {
             avifDiagnosticsPrintf(diag, "avifImageScale requested invalid width scale for libyuv [%u -> %u]", srcWidth, dstWidth);
             return AVIF_FALSE;
         }
-        const int fixedPointDivHeight = (int)(((int64_t)(srcHeight) << 16) / dstHeight);
-        const int64_t maxFixedValueHeight = (int64_t)fixedPointDivHeight * dstHeight;
-        if (maxFixedValueHeight > INT_MAX) {
+        if (srcHeight > 16384) {
             avifDiagnosticsPrintf(diag, "avifImageScale requested invalid height scale for libyuv [%u -> %u]", srcHeight, dstHeight);
             return AVIF_FALSE;
         }
```

Comment:
I believe these `(srcWidth > 16384)` and `(srcHeight > 16384)` tests could be merged and moved to line 72, to avoid code duplication and unnecessary work. `if (srcYUVPlanes[0] || srcAlphaPlane)` can be prepended to keep the same behavior.

Reply:
Done. Thanks for the suggestion!

Note: I believe srcYUVPlanes[0] cannot be possibly be NULL, because an AVIF image must have at least a Y plane. So the if (srcYUVPlanes[0]) test could be omitted. But I did not clean this up in this CL.